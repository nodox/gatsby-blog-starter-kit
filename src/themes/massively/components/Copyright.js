import React from 'react';


export default class Copyright extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div id="copyright">
        <ul>
          <li>{this.props.data.owner}</li>
          <li>Design: <a href="https://html5up.net">HTML5 UP</a></li>
          <li>Powered by: <a href="https://www.gatsbyjs.org/">GatsbyJS</a></li>
        </ul>
      </div>
    );
  }
}
