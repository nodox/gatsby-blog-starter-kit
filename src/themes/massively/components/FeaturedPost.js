import React from 'react';
import Link from 'gatsby-link';

export default class FeaturedPost extends React.Component {
  render() {

    return (
      <article className="post featured">
        <header className="major">
          <span className="date">{post.frontmatter.date}</span>
          <h2><Link to={post.frontmatter.path}>{post.frontmatter.title}</Link></h2>
          <p>{post.excerpt}</p>
        </header>
        <Link to={post.frontmatter.path} className="image main"><img src="images/pic01.jpg" alt="" /></Link>
        <ul className="actions">
          <li><Link to={post.frontmatter.path} className="button big">Full Story</Link></li>
        </ul>
      </article>
    );
  }
}