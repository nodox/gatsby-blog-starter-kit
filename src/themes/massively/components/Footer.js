import React from 'react';

export default class Footer extends React.Component {

  render() {
    return (
      <footer id="footer">
        <section>
          <form method="post"
                name="contact"
                action="/thanks/"
                data-netlify="true"
                data-netlify-honeypot="bot-field"
                encType="application/x-www-form-urlencoded">
            <input type="hidden" name="form-name" value="contact" />
            <p hidden>
             <label>
               Donâ€™t fill this out: <input name="bot-field" />
             </label>
            </p>
            <div className="field">
              <label>Name</label>
              <input type="text" name="name" id="name" placeholder="Your name"/>
            </div>
            <div className="field">
              <label>Email</label>
              <input id="email" type="email" name="email" placeholder="Your email"/>
            </div>
            <div className="field">
              <label>Message</label>
              <textarea id="message" name="message" rows="3" placeholder="Your message"></textarea>
            </div>
            <ul className="actions">
              <li><input type="submit" value="Send Message"/></li>
            </ul>
          </form>
        </section>
        <section className="split contact">
          <section>
            <h3>Social</h3>
            <ul className="icons alt">
              <li><a href="https://www.twitter.com/stevennatera" className="icon alt fa-twitter"><span className="label">Twitter</span></a></li>
              <li><a href="https://www.linkedin.com/in/snatera" className="icon alt fa-linkedin"><span className="label">Facebook</span></a></li>
              <li><a href="https://www.github.com/nodox" className="alt icon fa-github"><span className="label">GitHub</span></a></li>
            </ul>
          </section>
        </section>
      </footer>
    );
  }
}







