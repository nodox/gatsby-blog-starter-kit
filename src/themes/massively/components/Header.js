import React from 'react';
import Link from 'gatsby-link';

export default class Header extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <header id="header">
        <Link to="/" className="logo">{this.props.name}</Link>
      </header>
    );
  }
}
