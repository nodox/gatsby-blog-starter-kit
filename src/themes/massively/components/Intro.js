import React from 'react';
import PropTypes from 'prop-types';

export default class Intro extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div id="intro" className="">
        <div className="img-container">
          <div className="img-size-sm img-intro img-circle"></div>
        </div>
        <h1>{this.props.name}</h1>
        <p>{this.props.description}</p>
        <ul className="actions">
          <li>
            <a href="#header" className="button icon solo fa-arrow-down scrolly">Continue</a>
          </li>
        </ul>
      </div>
    );
  }
}
