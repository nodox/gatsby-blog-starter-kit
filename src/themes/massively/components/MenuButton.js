import React from 'react';
import Link from 'gatsby-link';

export class MenuButton extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <a href="#navPanel"
        id="navPanelToggle"
        className="alt"
        onClick={() => this.props.openPanel()}>Menu</a>
    );
  }
}
