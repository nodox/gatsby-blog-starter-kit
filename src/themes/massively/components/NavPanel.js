import React from 'react';
import Link from 'gatsby-link';

export default class NavPanel extends React.Component {

  constructor() {
    super();
  }

  closePanel(scrollToElement = false) {
    this.props.closePanelHandler()

    // if (scrollToElement) {
    //   // Scroll to a certain element
    //   document.querySelector('.major').scrollIntoView({ 
    //     behavior: 'smooth' 
    //   });
    // }
  }

  render() {
    return (
      <div id="navPanel">
        <nav>
          <ul className="links">
            <li><Link to="/" onClick={() => this.closePanel() }>Home</Link></li>
            <li><Link to="/about" onClick={() => this.closePanel() }>About</Link></li>
            <li><Link to="/speaking"onClick={() => this.closePanel() }>Speaking</Link></li>
            <li><Link to="/contact" onClick={() => this.closePanel() }>Contact</Link></li>
          </ul>
          <ul className="icons alt">
            <li><a href="https://www.twitter.com/stevennatera" className="icon fa-twitter alt"><span className="label">Twitter</span></a></li>
            <li><a href="https://www.linkedin.com/in/snatera" className="icon fa-linkedin alt"><span className="label">Facebook</span></a></li>
            <li><a href="https://www.github.com/nodox" className="icon fa-github alt"><span className="label">GitHub</span></a></li>
          </ul>
        </nav>
        <a href="#navPanel" className="close" onClick={() => this.closePanel() } style={{ WebkitTapHighlightColor: 'rgba(0, 0, 0, 0)'}}></a>
      </div>
    );
  }
}