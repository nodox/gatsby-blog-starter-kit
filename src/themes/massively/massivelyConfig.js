module.exports = {
    initialLinks: {
      paths: [
        {
          path: '/',
          title: 'Home',
        },
        {
          path: '/about',
          title: 'About'
        },
        {
          path: '/speaking',
          title: 'Speaking'
        },
        {
          path: '/contact',
          title: 'Contact'
        },
      ],
      socialLinks: [
        {
          path: 'https://www.twitter.com/stevennatera',
          icon: 'fa-twitter',
          name: 'Twitter',
        },
        {
          path: 'https://www.linkedin.com/in/snatera',
          icon: 'fa-linkedin',
          name: 'LinkedIn',
        },
        {
          path: 'https://www.github.com/nodox',
          icon: 'fa-github',
          name: 'Github',
        }
      ]
    },
    copyright: {
      owner: 'Â© Steven Natera',
    },
    introText: {
      '/': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/about': 'Hi! I\'m Steven Natera. I write this blog, teach software engineering workshops, and explore emerging technologies.',
      '/speaking': 'I deliver talks on NodeJS and I teach software engineering workshops.',
      '/contact': 'I\'m pretty easy to get hold of, here\'s how I use different channels to communicate with people and how best to contact me'
    }
}
