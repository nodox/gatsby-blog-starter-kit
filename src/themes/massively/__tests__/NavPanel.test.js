import React from 'react';
import renderer from 'react-test-renderer';
import { NavPanel } from '../components/NavPanel';
import * as config from '../massivelyConfig';

describe('<NavPanel />', () => {
  const component = shallow(
    <NavPanel navLinks={config.initialLinks} />
  );

  const props = component.instance().props;

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    expect(props).toHaveProperty('navLinks');
    expect(props).toHaveProperty('navLinks.socialLinks');
    expect(props).toHaveProperty('navLinks.paths');
  });

});
