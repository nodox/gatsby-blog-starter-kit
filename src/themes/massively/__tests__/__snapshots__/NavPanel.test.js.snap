// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavPanel /> should render correctly 1`] = `
<div
  id="navPanel"
>
  <nav>
    <ul
      className="links"
    >
      <li
        key="0"
      >
        <GatsbyLink
          onClick={[Function]}
          to="/"
        >
          Home
        </GatsbyLink>
      </li>
      <li
        key="1"
      >
        <GatsbyLink
          onClick={[Function]}
          to="/about"
        >
          About
        </GatsbyLink>
      </li>
      <li
        key="2"
      >
        <GatsbyLink
          onClick={[Function]}
          to="/speaking"
        >
          Speaking
        </GatsbyLink>
      </li>
      <li
        key="3"
      >
        <GatsbyLink
          onClick={[Function]}
          to="/contact"
        >
          Contact
        </GatsbyLink>
      </li>
    </ul>
    <ul
      className="icons alt"
    >
      <li
        key="0"
      >
        <a
          className="icon fa-twitter alt"
          href="https://www.twitter.com/stevennatera"
        >
          <span
            className="label"
          >
            Twitter
          </span>
        </a>
      </li>
      <li
        key="1"
      >
        <a
          className="icon fa-linkedin alt"
          href="https://www.linkedin.com/in/snatera"
        >
          <span
            className="label"
          >
            LinkedIn
          </span>
        </a>
      </li>
      <li
        key="2"
      >
        <a
          className="icon fa-github alt"
          href="https://www.github.com/nodox"
        >
          <span
            className="label"
          >
            Github
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <a
    className="close"
    href="#navPanel"
    onClick={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      }
    }
  />
</div>
`;
